lektor/admin/modules/static/livereload/worker.js

```diff
@@ -0,0 +1,112 @@
+/* eslint-env worker */
+"use strict";
+
+let eventsPath = null;
+let port = null;
+let currentVersionId = null;
+let eventSource = null;
+
+addEventListener("connect", (event) => {
+  // Only keep one active port, for whichever tab was last loaded.
```

Comment:
Since the worker script is loaded by `SharedWorker` and there is no template rendering there, we are not able to do so(at least with little effort). The events URL will be passed to the worker via message post and the worker can compare the URL with the one stored previously to determine whether to re-initialize the event source(In case you changed the URL and don't want to refresh the page).

Reply:
It's not hard to serve `worker.js` from a flask view which renders a template.
```py
@bp.route("/events/worker.js")
def worker_js():
    response = make_response(render_template("worker.js", events_url=url_for("livereload.events")))
    response.content_type = `"application/javascript"`
    return response
```